DXMainClient/DXGUI/Multiplayer/CnCNet/CnCNetLobby.cs

```diff
@@ -645,8 +649,17 @@ private string CanJoinGameByIndex(int gameIndex)
             if (ProgramConstants.IsInGame)
                 return "Cannot join game while the main game executable is running.";
 
-            HostedCnCNetGame hg = (HostedCnCNetGame)lbGameList.Items[gameIndex].Tag;
-
+            return null;
+        }
+        
+        /// <summary>
+        /// Determines whether or not a game is join-able.
+        /// If not, an error message is returned. Otherwise, null.
+        /// </summary>
+        /// <param name="hg"></param>
+        /// <returns></returns>
+        private string CanJoinGame(HostedCnCNetGame hg)
```

Comment:
I am not actually sure the method's code does what's written in method name. I mean, I would assume something named `CanJoinGame` should simply return a `bool`.

Reply:
The code using it seems kinda counterintuitive because of that too.

Reply:
I can update this method name, if you want. I was just modeling it after the other index based method of `private string CanJoinGameByIndex(int gameIndex)` that already exists. As my first PR, I didn't want to introduce more risk than necessary by making more changes than I had to, while also balancing matching existing naming conventions.

Reply:
perhaps they become `GetJoinGameErrorByIndex(..)` and `GetJoinGameError(..)`?

Reply:
@Metadorius see latest commit

Reply:
@Metadorius I'm glad you made me look at this a little closer. I believer there may have been some issues with the previous code in that the index based version was doing some non-index based checks that need to be considered for this function as well. I've added another commit that creates a base function to check global states and is called by both the index based version and the `HostedCnCNetGame` based version.